### PR TITLE
fix(cli): Add output file option to warp & core apply

### PR DESCRIPTION
**Pull request by ltyu**
### Description
Add --output flag to output file in `hyperlane warp & core apply`

### Backward compatibility
Yes

### Testing
Manual

### CODE REVIEW

File: typescript/cli/src/commands/warp.ts [hunk 1]
```diff
@@ -79,25 +84,47 @@ export const apply: CommandModuleWithWriteContext<{
       ...warpCoreConfigCommandOption,
       demandOption: false,
     },
+    out: outputFileCommandOption(
+      undefined,
+      false,
+      'The path to output a warp route deployment config JSON or YAML file.',
+    ),
```
Comment (jmrossy) [inline typescript/cli/src/commands/warp.ts:91]: Is it confusing that the output will automatically get added to the registries but also optionally this file?

**Review by jmrossy — APPROVED** [review]
Approving but with one concern. I'll leave the product decisions to you, Noah, and Connor so do whatever you think is best

File: typescript/cli/src/commands/warp.ts (hunk 1 above)
Comment (ltyu) [reply typescript/cli/src/commands/warp.ts:91]: Sorry, I should've marked this as draft as it was not ready. Still some discussion about the requirements of this

**Comment by ltyu** [comment]
Closing this because the intention was to update WarpCore artifacts upon `owner` update. however, the artifacts do not have an `owner` so it is not relevant.

**Pull request closed without merging**